File: Majority element.cpp
//Problem Link: https://leetcode.com/problems/majority-element/

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int,int> mp;
        for(int i=0;i<nums.size();i++)
            mp[nums[i]]+=1;
        int num=0;
        for(auto it:mp){
            if(it.second>nums.size()/2){
                num=it.first;
                break;
            }
        }
        return num;
    }
};

File: Majority element.js
//https://leetcode.com/problems/majority-element/


/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let a={};
    for(let i=0;i<nums.length;i++) {
        if(a[nums[i]]!=undefined)
            a[nums[i]]+=1;
        else
            a[nums[i]]=1
    }
    res=0,num=-1;
    console.log(a);
    Object.keys(a).forEach((i)=>{
        // console.log(i);
        if(a[i]>res)
            res=a[i],num=i;
    });
    return num;
};